menas/ui/generic/functions.js

```diff
@@ -13,34 +13,45 @@
  * limitations under the License.
  */
 
-var Functions = new function() {
-	this.urlBase = ""
+var Functions = new function () {
+  this.urlBase = "";
 
+  this.csrfHeader = "X-CSRF-TOKEN";
 
-	this.ajax = function(sPath, sMethod, oData, fnSuccess, fnError, oControl) {
-		if(oControl) oControl.setBusy(true);
-		
-		var oFormattedData = null;
-		if((sMethod.toLowerCase() === "post" || sMethod.toLowerCase() === "put") && typeof(oData) === "object") {
-			oFormattedData = JSON.stringify(oData)
-		} else oFormattedData = oData;
-		
-		$.ajax(this.urlBase + sPath, {
-			beforeSend : function(oJqXHR, oSettings) {
-				var token = $("meta[name='_csrf']").attr("content");
-				var header = $("meta[name='_csrf_header']").attr("content");
-				console.log("CRSF: " + header + " -> " + token)
-				oJqXHR.setRequestHeader(header, token);
-			},
-			complete: function() {
-				if(oControl) oControl.setBusy(false)
-			},
-			data: oFormattedData,
-			dataType : "json",
-			contentType: "application/json",
-			method : sMethod,
-			success : fnSuccess,
-			error : fnError
-		})
-	}
+  this.ajax = function (sPath, sMethod, oData, fnSuccess, fnError, oControl) {
+    if (oControl) oControl.setBusy(true);
+
+    let oFormattedData = null;
+    if ((sMethod.toLowerCase() === "post" || sMethod.toLowerCase() === "put")
+      && (typeof oData === "object")) {
```

Comment:
I think `&&` should be the last thing on the previous line. So the programmer reading it know that it is continuing from a first glance.

Reply:
I tend to disagree, for the same reason method calls on new lines start with a dot (e.g. `.map(...)`) makes it clear that this line isn't a statement on its own. I look at the start of lines a lot more than their ends of I'm just scanning the code.

Reply:
As this is subjective, dealer call. Just know, that people who look at the end exist 😄